lib/snowplow_tracker/emitters/cache.ex

```diff
@@ -0,0 +1,76 @@
+defmodule SnowplowTracker.Emitters.Cache do
+  @moduledoc """
+  Module which implements functions to interact with ETS
+  """
+
+  require Logger
+
+  alias :ets, as: Ets
+
+  @table Application.get_env(:snowplow_tracker, :table)
+  #@persist_every Application.get_env(:snowplow_tracker, :persist_every, 5000)
+  @path Application.get_env(:snowplow_tracker, :path, ".")
+  @lock "lock"
+
+  def init(table \\ @table) do
+    PersistentEts.new(
+      table,
+      "#{@path}/#{Atom.to_string(table)}.tab",
+      [:public, :set, :named_table, {:read_concurrency, true}]
+    )
+
+    Logger.log(:debug, "Table #{table} created successfully!")
+    {:ok, :success}
+  end
+
+  def insert(payload, table \\ @table) do
+    Ets.insert(table, payload)
+    {:ok, :success}
+  end
+
+  def match(table \\ @table) do
+    {
+      :ok,
+      Ets.match(table, {:"$1", :"$2", :"$3"})
+    }
+  end
+
+  def check_lock(table \\ @table) do
+    {
+      :ok,
+      Ets.lookup(table, @lock)
+    }
+  end
+
+  # coveralls-ignore-start
+  def release_lock(payload, table \\ @table)
+  # coveralls-ignore-stop
+
+  def release_lock({:ok, _msg}, table), do: delete_key(@lock, table)
+
+  def release_lock({:error, _}, _table), do: :ok
+
+  def set_lock(table \\ @table) do
+    try do
+      Ets.insert(table, {@lock, true})
+      Logger.log(:debug, "#{__MODULE__}: Lock set successfully!")
+      {:ok, :success}
+    rescue
+      _e in ArgumentError ->
```

Comment:
In which case `Ets.insert` will return this error? Why aren't we handling this error here? - 

https://github.com/scripbox/snowplow-elixir-tracker/pull/12/files#diff-e35453d088eb870224a7284feb5d8710R26

Reply:
It will return an error if:
- The table does not exist.
- The key is already present.